src/main/scala/caravan/bus/common/DummyMemory.scala

```diff
@@ -0,0 +1,55 @@
+package caravan.bus.common
+import chisel3._
+import chisel3.experimental.ChiselEnum
+import chisel3.stage.ChiselStage
+import chisel3.util.{Cat, Decoupled, MuxLookup}
+import chisel3.util.experimental.loadMemoryFromFile
+
+
+//implicit parameters for Config, Request and Response
+class DummyMemController(programFile: Option[String])(implicit val config: BusConfig, implicit val request: AbstrRequest, implicit val response: AbstrResponse) extends Module {
+    val io = IO(new Bundle {
+        val req = Flipped(Decoupled(request))
+        val rsp = Decoupled(response)
+    })
+
+    val validReg = RegInit(false.B)
+
+    io.rsp.valid := validReg
+    io.rsp.bits.error := false.B
+    io.req.ready := true.B
+
+    // masked memory init
+    val mem = SyncReadMem(1024, Vec(4, UInt(8.W)))
+
+    if (programFile.isDefined) {
+        loadMemoryFromFile(mem, programFile.get)
+    }
+
+    
+
+    when(io.req.bits.isWrite){
```

Comment:
A better approach would be to use `when(io.req.fire() && io.req.bits.isWrite)` . The `fire()` method on decoupled interfaces is a handy method that is equivalent to `when(io.req.valid && io.req.ready)` which means that we only write into the memory when the controller is __ready__ to accept data and when the requestor is sending a __valid__ data.

Reply:
Take hint from here: https://github.com/merledu/jigsaw/blob/main/src/main/scala/jigsaw/rams/fpga/BlockRam.scala#L131-L143